From eeea155ca85e6098e8859f8cf68b8cdc1904a53f Mon Sep 17 00:00:00 2001
From: "Souptick Joarder (HPE)" <jrdr.linux@gmail.com>
Date: Sat, 19 Feb 2022 22:09:15 +0530
Subject: [PATCH 350/921] [Backport] bpf: Initialize ret to 0 inside
 btf_populate_kfunc_set()

mainline inclusion
from mainline-v5.18-rc1
commit d0b3822902b6af45f2c75706d7eb2a35aacab223
category: bugfix
bugzilla: N/A

Reference: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=d0b3822902b6af45f2c75706d7eb2a35aacab223

------------------------------------------------------

Kernel test robot reported below error ->

kernel/bpf/btf.c:6718 btf_populate_kfunc_set()
error: uninitialized symbol 'ret'.

Initialize ret to 0.

Fixes: dee872e124e8 ("bpf: Populate kfunc BTF ID sets in struct btf")
Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Souptick Joarder (HPE) <jrdr.linux@gmail.com>
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Acked-by: Kumar Kartikeya Dwivedi <memxor@gmail.com>
Link: https://lore.kernel.org/bpf/20220219163915.125770-1-jrdr.linux@gmail.com

Conflicts:
	kernel/bpf/btf.c

Signed-off-by: Yang Jihong <yangjihong1@huawei.com>
---
 kernel/bpf/btf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/bpf/btf.c b/kernel/bpf/btf.c
index 75436e370a06..1062493348bc 100644
--- a/kernel/bpf/btf.c
+++ b/kernel/bpf/btf.c
@@ -5680,7 +5680,7 @@ static int __btf_populate_kfunc_set(struct btf *btf, enum btf_kfunc_hook hook,
 	struct btf_kfunc_set_tab *tab;
 	struct btf_id_set *set;
 	u32 set_cnt;
-	int ret;
+	int ret = 0;
 
 	if (hook >= BTF_KFUNC_HOOK_MAX || type >= BTF_KFUNC_TYPE_MAX) {
 		ret = -EINVAL;
-- 
2.17.1

